### PR TITLE
app-editors/hyx: new package, add 2021.06.09

### PR DESCRIPTION
**Pull request by hamarituc**
Signed-off-by: Mario Haustein <mario.haustein@hrz.tu-chemnitz.de>

### CODE REVIEW

**Comment by thesamesam** [comment]
```
>>> Compiling source in /var/tmp/portage/app-editors/hyx-2021.06.09/work/hyx-2021.06.09 ...
make -j32
cc \
        -Werror=format-security -Werror=implicit-function-declaration -Wformat -O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches     -ggdb3 -Werror=format-security -Werror=implicit-function-declaration -Wformat -std=c99 -pedantic -Wall -Wextra -DNDEBUG \
        hyx.c common.c blob.c history.c view.c input.c \
        -o hyx
>>> Source compiled.
>>> Test phase [not enabled]: app-editors/hyx-2021.06.09

>>> Install app-editors/hyx-2021.06.09 into /var/tmp/portage/app-editors/hyx-2021.06.09/image
>>> Completed installing app-editors/hyx-2021.06.09 into /var/tmp/portage/app-editors/hyx-2021.06.09/image

 * Final size of build directory: 364 KiB
 * Final size of installed tree:  272 KiB


 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/hyx
```

* Doesn't respect CC
* Doesn't respect LDFLAGS (you need e.g. `LDFLAGS="${LDFLAGS} -Wl,--defsym=__gentoo_check_ldflags__=0` to get this)

File: app-editors/hyx/hyx-2021.06.09.ebuild [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+EAPI=8
+
+DESCRIPTION="a minimalistic console hex editor with vim-like controls"
+HOMEPAGE="https://yx7.cc/code/"
+SRC_URI="https://yx7.cc/code/hyx/${P}.tar.xz"
+
+LICENSE="MIT-with-advertising"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (thesamesam) [inline app-editors/hyx/hyx-2021.06.09.ebuild:13]: drop IUSE, DEPEND, RDEPEND

File: app-editors/hyx/hyx-2021.06.09.ebuild [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="a minimalistic console hex editor with vim-like controls"
```
Comment (thesamesam) [inline app-editors/hyx/hyx-2021.06.09.ebuild:6]:
```suggestion
DESCRIPTION="A minimalistic console hex editor with vim-like controls"
```

**Comment by thesamesam** [comment]
Eyeballing it, it looks like it'll still not respect CC, right?

**Comment by hamarituc** [comment]
> Eyeballing it, it looks like it'll still not respect CC, right?
> 
> (If want to find these bugs other than by-eye in build.log, try binutils-config[-native-symlinks], ditto gcc-config: https://wiki.gentoo.org/wiki/Project:Toolchain/use_native_symlinks)

Sorry, you wrote `CC` but I read `CFLAGS` instead and was wondering about that, since `CFLAGS` is respected.

I learned now `$(CC)` isn't defined in the build environment. I still have to learn a lot ;). I will fix it. Sorry for bothering.

**Comment by thesamesam** [comment]
> > Eyeballing it, it looks like it'll still not respect CC, right?
> > (If want to find these bugs other than by-eye in build.log, try binutils-config[-native-symlinks], ditto gcc-config: [wiki.gentoo.org/wiki/Project:Toolchain/use_native_symlinks](https://wiki.gentoo.org/wiki/Project:Toolchain/use_native_symlinks))
> 
> Sorry, you wrote `CC` but I read `CFLAGS` instead and was wondering about that, since `CFLAGS` is respected.
> 
> I learned now `$(CC)` isn't defined in the build environment. I still have to learn a lot ;). I will fix it. Sorry for bothering.

Please don't be afraid to ask things! I want you to -- it's how we'll help you learn.

Yeah, this is one of those things which often surprises people, and I remember being confused by it at first. The way most packages get it right _without_ using `tc-export` is that the configure script just looks for ${CHOST}-gcc and so on and has fallbacks, then if the user sets CC in the env, the configure script will pick up on that.

**Pull request closed without merging**